php-checks/src/test/resources/checks/HardCodedIpAddressCheck.php

```diff
@@ -0,0 +1,17 @@
+<?php
+
+const serverAddress = "1.20.33.145";  // Noncompliant {{Make sure using this hardcoded IP address is safe here.}}
+//                    ^^^^^^^^^^^^^
+const nonRelatedValue = 42;
+
+class A {
+  const constFieldNameWithPasswordInIt = "8.8.8.8"; // Noncompliant
+  const otherConstFieldName = "";
+}
+
+function foo($socket) {
+  static $otherStaticVariableName = "xxx";
+  socket_connect($socket, '1.1.1.1', 23); // Noncompliant
+}
+
+const localhost = "127.0.0.1"; // Compliant - exception for localhost
```

Comment:
There is not test where an IP address is only a part of a literal, e.g. `'http://8.8.8.8/something'`